apps/api/src/service/ApplicationStateMachine.ts

```diff
@@ -0,0 +1,23 @@
+import {  ApplicationStates, ApplicationStateValues } from "@pcgl-daco/data-model/src/types.js";
```

Comment:
a) new files need the Copyright at the top
b) We have ApplicationStateManager defined at https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/src/api/states.ts#L61
which extends functionality https://github.com/eram/typescript-fsm

including the `.can` method which could essentially perform the same function as this.
I would use the existing class and work within that framework.

Now, there may not be implementation examples for the State Machine yet in the API, except for in the unit tests: https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/tests/api/application-state.test.ts

So I can review this with you and show how Jon and I were intending to implement the state machine and action updates